lib/amqp/client/message_coding.rb

```diff
@@ -0,0 +1,100 @@
+# frozen_string_literal: true
+
+require "zlib"
+require "stringio"
+require "json"
+
+module AMQP
+  class Client
+    # Methods for encoding/decoding message bodies
+    module MessageCoding
+      private
+
+      def parse_body(body, properties)
+        decoded_data = decode_body(body, properties)
+        case properties.content_type
+        when "application/json" then parse_json(decoded_data)
+        when "text/plain", "", nil then decoded_data
+        else raise Error::UnsupportedContentType, properties.content_type
+        end
+      end
+
+      def decode_body(body, properties)
+        case properties.content_encoding
+        when "gzip"
+          decode_gzip(body)
+        when "deflate"
+          decode_deflate(body)
+        when "", nil
+          body
+        else
+          raise Error::UnsupportedContentEncoding, properties.content_encoding
+        end
+      end
+
+      def encode_body(body, properties)
+        body = serialize_body(body, properties)
+        case properties[:content_encoding]
+        when "gzip"
+          encode_gzip(body)
+        when "deflate"
+          encode_deflate(body)
+        when "", nil
+          body
+        else
+          raise Error::UnsupportedContentEncoding, properties[:content_encoding]
```

Comment:
Could it make sense to return the raw data in this case instead (to avoid a breaking change)? Let's say someone uses something exotic as brotli as "content encoding", then they would pass through here to see a raise (there are other "common" ones: https://developer.mozilla.org/en-US/docs/Web/HTTP/Reference/Headers/Content-Encoding#directives).

Client user could work around it by not setting the content_encoding header, but that may cause issues in the other end instead.

Another option is to make the raise configurable or even possible to register their own encoders.